src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/BuildWithComponentsIntegrationTest.cs

```diff
@@ -93,9 +93,9 @@ private async Task Build_ComponentsWorks(MSBuildProcessKind msBuildProcessKind)
             Assert.AssemblyDoesNotContainType(result, Path.Combine(OutputPath, "MvcWithComponents.Views.dll"), "AspNetCore.Components_Counter");
 
             // Verify a regular View appears in the views dll, but not in the main assembly.
-            Assert.AssemblyDoesNotContainType(result, Path.Combine(OutputPath, "MvcWithComponents.dll"), "AspNetCore.Views.Home.Index");
+            Assert.AssemblyDoesNotContainType(result, Path.Combine(OutputPath, "MvcWithComponents.dll"), "MvcWithComponents.Views.Home.Views_Home_Index");
             Assert.AssemblyDoesNotContainType(result, Path.Combine(OutputPath, "MvcWithComponents.dll"), "AspNetCore.Views_Home_Index");
-            Assert.AssemblyContainsType(result, Path.Combine(OutputPath, "MvcWithComponents.Views.dll"), "AspNetCore.Views_Home_Index");
+            Assert.AssemblyContainsType(result, Path.Combine(OutputPath, "MvcWithComponents.Views.dll"), "MvcWithComponents.Views.Home.Views_Home_Index");
```

Comment:
This is because non-component web apps also use `RootNamespace` as their base namespace after this change. Are we okay with this behavior change?

Reply:
This is a breaking change - I think it's worth considering because it's an improvment
1. Discuss with @danroth27 
2. Mark PR as breaking-change
3. Write an announcement post
4. Make sure you're ready to react to this change in aspnet/AspNetCore if necessary

Reply:
Not doing the behavior change is also an option if we special case the file kind in the namespace calculation.

Reply:
The reason why this is a breaking change is that it will change the namespace resolution order for things that are in your app.

Reply:
Discussed it with @danroth27. This might not have a lot of value for non-component apps. Going to special case it with file kind.